example/cpp/tests/assert.hpp

```diff
@@ -0,0 +1,14 @@
+#include<iostream>
+#include<cstdlib>
+
+#define assert_eq(name, lhs, rhs) \
+    if (lhs != rhs) { \
+        std::cout << name << ": Expected " #lhs " == " #rhs ", found " << lhs << std::endl; \
```

Comment:
suggestion: It might be nice to have the source file and line here, e.g. using `__FILE__` and` __LINE__` or the C++20 std::source_location. I'm actually not sure how these work inside a macro, if you have to pass them in as arguments it might not be worth the extra clutter.